### PR TITLE
supervisor-create-subject

### PR DESCRIPTION
**Pull request by hienbx94**
I have created pull request. Please check it

### CODE REVIEW

**Comment by ttuan** [comment]
OK

**Comment by trantube94** [comment]
ok

File: app/assets/javascripts/nested_forms.js.coffee [hunk 1]
```diff
@@ -0,0 +1,23 @@
+jQuery ->
+  $(".duplicatable_nested_form").click (e) ->
+    e.preventDefault()
+
+    lastNestedForm = $(".duplicatable_form").last()
+    newNestedForm  = $(lastNestedForm).clone()
+    formsOnPage    = $(".duplicatable_form").length
```
Comment (ThuBM) [inline app/assets/javascripts/nested_forms.js.coffee]: indent

File: app/assets/javascripts/nested_forms.js.coffee [hunk 2]
```diff
@@ -0,0 +1,23 @@
+jQuery ->
+  $(".duplicatable_nested_form").click (e) ->
+    e.preventDefault()
+
+    lastNestedForm = $(".duplicatable_form").last()
+    newNestedForm  = $(lastNestedForm).clone()
```
Comment (ThuBM) [inline app/assets/javascripts/nested_forms.js.coffee]: indent

File: app/assets/javascripts/nested_forms.js.coffee [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    $(newNestedForm).find("select, input").each ->
+      oldId = $(this).attr "id"
+      newId = oldId.replace(new RegExp(/_[0-9]+_/), "_#{formsOnPage}_")
+      $(this).attr "id", newId
+
+      oldName = $(this).attr "name"
+      newName = oldName.replace(new RegExp(/\[[0-9]+\]/), "[#{formsOnPage}]")
+      $(this).attr "name", newName
+
+    $( newNestedForm ).insertAfter( lastNestedForm )
```
Comment (ThuBM) [inline app/assets/javascripts/nested_forms.js.coffee]: remove space before ), after (

File: app/controllers/admin/subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @task = @subject.tasks.build
+  end
+
+  def create
+    @subject = Subject.new subject_params
+    if @subject.save
+      flash[:success] = t "subject_success"
+      redirect_to @subject
+    else
+      render "new"
```
Comment (ThuBM) [inline app/controllers/admin/subjects_controller.rb]: flash

**Comment by ThuBM** [comment]
OK

**Comment by trantube94** [comment]
ok

File: app/controllers/admin/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      redirect_to admin_subjects_path 
+    else
+      render "new"
+    end
+  end
+
+  private
+  def subject_params
+    params.require(:subject).permit :name, :description, :day_work,
+      tasks_attributes: [:id, :name, :description]
```
Comment (chuanhtuan) [inline app/controllers/admin/subjects_controller.rb]: _destroy

**Pull request merged**